onnx/defs/tensor/defs.cc

```diff
@@ -3890,4 +3890,78 @@ ONNX_OPERATOR_SET_SCHEMA(
 	        }
         }));
 
+static const char* Col2Im_ver16_doc = R"DOC(
+The operator rearranges the input matrix columns into blocks.
+
+Col2Im behaves like PyTorch's fold for two output spatial dimensions https://pytorch.org/docs/stable/generated/torch.nn.Fold.html
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+	Col2Im,
+	16,
+	OpSchema()
+	.Attr(
+		"dilation",
+		"Dilation value along each spatial axis of the kernel. "
+    "If not present, the dilation defaults to 1 for each spatial axis",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.Attr(
+		"padding",
+		"Padding for the beginning and ending along each spatial axis, it can take any value greater than or equal to 0. "
+    "The value represent the number of pixels added to the beginning and end part of the corresponding axis. "
+    "`padding` format should be as follow [x1_begin, x2_begin...x1_end, x2_end,...], where xi_begin the number of "
+    "pixels added at the beginning of axis `i` and xi_end, the number of pixels added at the end of axis `i`. "
+    "If not present, the padding defaults to 0 along start and end of each spatial axis.",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.Attr(
+		"stride",
+		"Stride along each spatial axis. "
+    "If not present, the stride faults to 1 for each spatial axis.",
+    AttributeProto::INTS,
+    OPTIONAL_VALUE)
+	.SetDoc(Col2Im_ver16_doc)
+	.Input(
+		0,
+		"input",
+		"Input data tensor to be rearranged into blocks.",
+		"T",
+		OpSchema::Single,
+		true,
+		1,
+		OpSchema::Differentiable)
+	.Input(
+		1,
+		"output_shape",
```

Comment:
output_size in pytorch contains output [H, W] and not the entire output shape... is this the same for the output_shape in onnx? Add more clarification.

Reply:
I didn't understand this part. Could you elaborate?

Reply:
From: https://pytorch.org/docs/stable/generated/torch.nn.Fold.html
```
output tensor of shape (N, C, \text{output\_size}[0], \text{output\_size}[1], \dots)(N,C,output_size[0],output_size[1],…)
.
.
.
output_size (int or tuple) – the shape of the spatial dimensions of the output **(i.e., output.sizes()[2:])**
```

where N and C come from:
```
Consider a batched input tensor containing sliding local blocks, e.g., patches of images, of shape (N, C \times \prod(\text{kernel\_size}), L)(N,C×∏(kernel_size),L), 
```